src/coreclr/jit/flowgraph.cpp

```diff
@@ -1270,6 +1270,44 @@ bool Compiler::fgCastNeeded(GenTree* tree, var_types toType)
     return true;
 }
 
+//-------------------------------------------------------------------------------------
+// fgCastRequiresHelper: Check whether a given cast must be converted to a helper call.
+//
+// Arguments:
+//    fromType - The source type of the cast.
+//    toType   - The target type of the cast.
+//    overflow - True if the cast has the GTF_OVERFLOW flag set.
+//
+// Return Value:
+//    True if the cast requires a helper call, otherwise false.
+//
+bool Compiler::fgCastRequiresHelper(var_types fromType, var_types toType, bool overflow /* false */)
+{
+    if (varTypeIsFloating(fromType))
+    {
+        return (varTypeIsIntegral(toType) && overflow)
```

Comment:
Do we need the `varTypeIsIntegral` check? We should never have a float->float cast that is marked for overflow, right?